examples/src/main/java/ai/timefold/solver/examples/common/experimental/impl/ComparableValue.java

```diff
@@ -0,0 +1,32 @@
+package ai.timefold.solver.examples.common.experimental.impl;
+
+record ComparableValue<Value_, Point_ extends Comparable<Point_>>(Value_ value, Point_ index)
+        implements
+            Comparable<ComparableValue<Value_, Point_>> {
+
+    @Override
+    public int compareTo(ComparableValue<Value_, Point_> other) {
+        if (this == other) {
+            return 0;
+        }
+        Point_ point1 = this.index;
+        Point_ point2 = other.index;
```

Comment:
This would previously do a hashmap lookup, which is extremely expensive for a comparator.